soundcard/pulseaudio.py

```diff
@@ -367,11 +354,54 @@ def record(self, numframes):
                     chunk = numpy.zeros(nbytes_ptr[0]//4, dtype='float32')
                 if nbytes_ptr[0] > 0:
                     self._pulse._pa_stream_drop(self.stream)
-                    captured_data.append(chunk)
-                    captured_frames += len(chunk)/self.channels
+                    return chunk
             else:
                 time.sleep(0.001)
-        return numpy.reshape(numpy.concatenate(captured_data), [-1, self.channels])
+
+    def record(self, numframes=None):
+        """Record some audio data.
```

Comment:
The docstring should be updated.

Maybe something along the lines of

> Record a block of audio data.
>
> The data will be returned as a `frames × channels` float32 numpy array.
>
> This function will wait until `numframes` frames have been recorded. If `numframes` is given, it will return exactly `numframes` frames, and buffer the rest for later. If `numframes` is `None`, it will return whatever the audio backend has available right now. Use this if latency must be kept to a minimum, but be aware that block sizes can change at the whims of the audio backend.